StyleCop.Analyzers/StyleCop.Analyzers.Test/OrderingRules/SA1217UnitTests.cs

```diff
@@ -0,0 +1,171 @@
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestValidUsingDirectivesInCompilationUnitAsync()
+        {
+            var testCode = @"using System;
+using static System.Array;
+using Execute = System.Action;
+using static System.Math;
+
+public class Foo
+{
+}
+";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the analyzer will produce the proper diagnostics when the using directives are ordered wrong.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestInvalidUsingDirectivesOrderingAsync()
+        {
+            var testCode = @"namespace Foo
+{
+    using System;
+    using Execute = System.Action;
+    using static System.Math;
+    using static System.Array;
+}
+
+namespace Bar
+{
+    using static System.Math;
+    using Execute = System.Action;
+    using static System.Array;
+    using System;
+}
+";
+
+            DiagnosticResult[] expectedDiagnostics =
+            {
+                this.CSharpDiagnostic().WithLocation(5, 5).WithArguments("System.Math", "System.Array"),
+                this.CSharpDiagnostic().WithLocation(11, 5).WithArguments("System.Math", "System.Array")
+            };
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expectedDiagnostics, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the analyzer will not produce a diagnostic when the using directives have inline comments.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestValidUsingDirectivesWithInlineCommentsAsync()
+        {
+            var testCode = @"namespace Foo
+{
+    using System;
+    using Execute = System.Action;
+    using static /* B */ System.Array;
+    using static /* A */ System.Math;
+}
+";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the analyzer will produce the proper diagnostics when one of the using directives has a global prefixes.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestInvalidUsingDirectivesWithGlobalPrefixAsync()
+        {
+            var testCode = @"namespace Foo
+{
+    using System;
+    using Execute = System.Action;
+    using static System.Math;
+    using static global::System.Array;
```

Comment:
Visual Studio does not have an option to automatically sort using directives ignoring the alias, which means we're going to have some "annoyances" if we ignore them. In particular, a solution-wide "organize using operation" would incorrectly order using directives in files containing aliases. On the other hand if we do consider the using alias then we'll have to make a decision regarding the impact of an alias on SA1208 (SystemUsingDirectivesMustBePlacedBeforeOtherUsingDirectives).